pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	if installStatus.CalicoVersion != components.EnterpriseRelease {
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	pullSecrets, err := utils.GetNetworkingPullSecrets(installation, r.client)
+	if err != nil {
+		reqLogger.Error(err, "Error retrieving pull secrets")
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error retrieving pull secrets", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error retrieving pull secrets", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Fetch any existing default FelixConfiguration object.
+	fc := &crdv1.FelixConfiguration{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: "default"}, fc)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading FelixConfiguration", err, reqLogger)
+		for _, egw := range egwsToReconcile {
+			r.setDegraded(ctx, &egw, "Error reading felix configuration", err)
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile all the EGWs
+	var errMsgs []string
+	for _, egw := range egwsToReconcile {
+		_, err = r.reconcileEgressGateway(ctx, &egw, reqLogger, variant, fc, pullSecrets, installation, unreadyEGW)
+		if err != nil {
+			reqLogger.Error(err, "Error reconciling egress gateway")
+			errMsgs = append(errMsgs, err.Error())
+		}
+	}
+	if len(errMsgs) != 0 {
+		return reconcile.Result{}, fmt.Errorf(strings.Join(errMsgs, ";"))
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileEgressGateway) setDegraded(ctx context.Context, egw *operatorv1.EgressGateway, msg string, err error) {
+	if err != nil {
+		setDegraded(r.client, ctx, egw, reconcileErr, fmt.Sprintf("%s Name = %s, Namespace = %s, err = %s", msg, egw.Name, egw.Namespace, err.Error()))
+	} else {
+		setDegraded(r.client, ctx, egw, reconcileErr, fmt.Sprintf("%s Name = %s, Namespace = %s", msg, egw.Name, egw.Namespace))
+	}
+}
+
+func (r *ReconcileEgressGateway) reconcileEgressGateway(ctx context.Context, egw *operatorv1.EgressGateway, reqLogger logr.Logger,
+	variant operatorv1.ProductVariant, fc *crdv1.FelixConfiguration, pullSecrets []*v1.Secret,
+	installation *operatorv1.InstallationSpec, unreadyEGW *operatorv1.EgressGateway) (reconcile.Result, error) {
+
+	preDefaultPatchFrom := client.MergeFrom(egw.DeepCopy())
+	// update the EGW resource with default values.
+	fillDefaults(egw, installation)
+	// Validate the EGW resource.
+	err := validateEgressGateway(ctx, r.client, egw)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error validating Egress Gateway Name = %s, Namespace = %s", egw.Name, egw.Namespace))
+		r.status.SetDegraded(operatorv1.ResourceValidationError,
```

Comment:
Can we simplify this setDegraded logic at all?

We have two functions in this file named `setDegraded`, and another one on the status manager, and it's not clear what each of them do since they don't have any comments, which also makes it hard to tell if they are being called in the correct places.

Maybe clarifying the names and adding comments would resolve this, but also wonder if any of them can be combined? 

I'll keep looking closer to see if I can come up with anything.

Reply:
Updating Name and Namespace of the EGW resource in its own condition is not needed. Hence removed r.setDegraded. Added comments as well.